patches/server/0362-Anti-Xray.patch

```diff
@@ -1608,6 +1608,30 @@ index 91ef1b0e06c30668fe4bfb18ecdf2fe499f72fee..36b7de78fa69f652079d74252286bb6d
      }
  
      @Override
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 706d5718997181279f7ec715526b4d8f2b6162a2..05142efad717698f72220fd8f421a2e28576c867 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -390,11 +390,17 @@ public class CraftWorld extends CraftRegionAccessor implements World {
+                 List<ServerPlayer> playersInRange = playerChunk.playerProvider.getPlayers(playerChunk.getPos(), false);
+                 if (playersInRange.isEmpty()) return;
+ 
+-                ClientboundLevelChunkWithLightPacket refreshPacket = new ClientboundLevelChunkWithLightPacket(chunk, this.world.getLightEngine(), null, null, true);
++                Map<Object, ClientboundLevelChunkWithLightPacket> cacheMap = new HashMap<>(); // Paper - Xray refreshChunk
+                 for (ServerPlayer player : playersInRange) {
+                     if (player.connection == null) continue;
+ 
+-                    player.connection.send(refreshPacket);
++                    // Paper start - Xray refreshChunk
++                    Boolean shouldModify = chunk.getLevel().chunkPacketBlockController.shouldModify(player, chunk);
++
++                    player.connection.send(cacheMap.computeIfAbsent(shouldModify, (s) -> { // Use  connection to prevent creating firing event
```

Comment:
Just noticed that there are two spaces in `// Use  connection ...`. Not sure how nitpicky we are. :D